modules/library/jdbc/src/main/java/org/geotools/jdbc/JDBCDataStore.java

```diff
@@ -1946,6 +1949,14 @@ private void insertPS(
         }
     }
 
+    private Geometry getGeometryAccordingtoBinding(Object value, Class<?> binding) {
+        Geometry g = (Geometry) value;
+        if (g != null && g instanceof MultiSurface && binding.equals(MultiPolygon.class)) {
```

Comment:
In general, no, this change is not acceptable. This class is shared by multiple database implementations, some might need linearization, others do not.

Please back out changes to the JDBCDataStore class, and implement linearization, if needed, in the Dialect instead. E.g., in the case of PostGIS PS dialect, in setGeometryValue, just before using a WKBWriter, but not in the non-PS path, where the WKT writer is able to write curved geometries.
In the context of the PostGIS example, a better fix would be to enable the WKB writer to write curved geometries instead. Even keeping linearization, I don't understand why linearization is needed only for MultiSurface, there are other curved geometry types, the WKB writer cannot deal with any of them... why not test for CurvedGeometry and linearize instead?

Finally, the change must be guarded by a new test (similar to the JDBCCurvesTest family, but for writing).

Reply:
Thanks for the review @aaime 

In fact, problem is also in non-PS dialect. 
In our case we have a feature stored in postgis with a geometry of type Multipolygon. 
When we execute a WFS transaction on this feature, JDBCDatastore encode as MultiSurface because GML schemas is mapped on this geometry type. In older version (<27.X), the MultiSurface geometry type was encoded as Multipolygon

I use the binding to get the type of the geometry feature and transform the transaction geometry to the right type . In non-PS dialect i don't have access to the binding

Reply:
Ok, I see what you mean. But still, the test is too specific, why linearize MultiSurface and not the other curved geometries? The test could be done using CurvedGeometry as the reference instead, and the method should be called "linearize", or something like that (that's what it actually does). With a 100 columns limit, we should try to have short yet evocative method names, if possible.